### ISSUE TITLE
Bug: Focalboard plugin server error when dragging multiple cards

### ISSUE THREAD

**Issue by chenilim** [mattermost/focalboard#650]
**Steps to reproduce the behavior:**
1. Set up an on-prem Mattermost v5.36 server
  a. Or register and login on the [test server I used](http://ec2-54-209-196-240.compute-1.amazonaws.com/signup_user_complete/?id=ab3o8ftp1ibw3cy7qs3hptxc9h)
2. install the Mattermost Plugin (0.6.7+)
3. Open Focalboard and add a Personal Tasks (or similar) template board
4. Drag cards to different columns - this works
5. Drag 3 or more cards to the same column
6. Open the Chrome Dev tools console (Cmd+Shift+I)
7. Press shift, click on the first card, then click on the last card to select all cards in the column
8. Drag 3+ cards to another column
9. The drag fails
10. Notice errors in the dev console

Repeat steps 7-8 if this doesn't repro.

**Expected behavior:**
This should work.

**Platform:**
- Server: 5.36-rc2
 - Browser and version: Chrome 91 on macOS 10.15.7

**Additional context:**
The following errors are seen in the System Console log:
```
{"level":"error","ts":1624988327.5582626,"caller":"mlog/log.go:251","msg":"Plugin failed to ServeHTTP, RPC call failed","plugin_id":"focalboard","error":"unexpected EOF"}
{"level":"error","ts":1624988327.5592828,"caller":"mlog/log.go:251","msg":"Plugin failed to ServeHTTP, RPC call failed","plugin_id":"focalboard","error":"unexpected EOF"}
{"level":"error","ts":1624988327.5596976,"caller":"http/server.go:3093","msg":"http: response.WriteHeader on hijacked connection from github.com/mattermost/mattermost-server/v5/plugin.(*hooksRPCClient).ServeHTTP (client_rpc.go:337)","source":"httpserver"}
{"level":"error","ts":1624988327.5605214,"caller":"http/server.go:3093","msg":"http: response.Write on hijacked connection from fmt.Fprintln (print.go:265)","source":"httpserver"}
{"level":"error","ts":1624988327.5608683,"caller":"mlog/log.go:251","msg":"Plugin failed to ServeHTTP, RPC call failed","plugin_id":"focalboard","error":"unexpected EOF"}
```

**Comment by chenilim** [comment]
@sbishel, this is a separate issue from the DB connections in 0.7.1, but might be related to observed failures of the 0.6.7 plugin that's been released. Please assign this as a P1 to investigate. I observed this failure on 2 separate servers (running 5.3.6 RC2 and the other RC1).

@ogi-m, are you able to repro this issue on 5.36 and/or 5.37?

**Comment by chenilim** [comment]
For the record, this also repros with the latest Focalboard 0.7.1 plugin against Mattermost 5.36 (RC1).

**Comment by harshilsharma63** [comment]
@chenilim I'm looking into this.

**Comment by harshilsharma63** [comment]
Setting Mattermost log level to debug brought out the true error -


```
DEBU[2021-06-30T12:01:42.971648+05:30] panic: concurrent write to websocket connection  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.97167+05:30]                                               caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971675+05:30] goroutine 173 [running]:                      caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.97168+05:30] github.com/gorilla/websocket.(*messageWriter).flushFrame(0xc000638150, 0x1, 0x0, 0x0, 0x0, 0xc0004e4900, 0x0)  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971684+05:30] 	/home/runner/go/pkg/mod/github.com/gorilla/websocket@v1.4.2/conn.go:610 +0x60e  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971692+05:30] github.com/gorilla/websocket.(*messageWriter).Close(0xc000638150, 0x203000, 0xc000c368d0)  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.9716961+05:30] 	/home/runner/go/pkg/mod/github.com/gorilla/websocket@v1.4.2/conn.go:724 +0x65  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971704+05:30] github.com/gorilla/websocket.(*Conn).beginMessage(0xc000966580, 0xc000b81650, 0x1, 0x100d850, 0xc000443180)  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.9717078+05:30] 	/home/runner/go/pkg/mod/github.com/gorilla/websocket@v1.4.2/conn.go:473 +0x262  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971711+05:30] github.com/gorilla/websocket.(*Conn).NextWriter(0xc000966580, 0x1, 0x80, 0x98, 0x2b00108, 0xc000443180)  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971715+05:30] 	/home/runner/go/pkg/mod/github.com/gorilla/websocket@v1.4.2/conn.go:513 +0x53  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971719+05:30] github.com/gorilla/websocket.(*Conn).WriteJSON(0xc000966580, 0x1cad3e0, 0xc000443180, 0xc000443180, 0x5)  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971723+05:30] 	/home/runner/go/pkg/mod/github.com/gorilla/websocket@v1.4.2/json.go:24 +0x49  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.9717278+05:30] github.com/mattermost/focalboard/server/ws.(*Server).BroadcastBlockChange(0xc0005540b0, 0xc00003a3d3, 0x1a, 0xc00053ec90, 0x24, 0xc00053ed50, 0x24, 0xc00053ed80, 0x24, 0xc00073cd80, ...)  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971733+05:30] 	/home/runner/work/focalboard/focalboard/server/ws/websockets.go:415 +0x5a5  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971738+05:30] github.com/mattermost/focalboard/server/app.(*App).InsertBlocks(0xc0008585a0, 0xc00003a3d3, 0x1a, 0xc0004e1d40, 0x1, 0x4, 0x0, 0x0)  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971742+05:30] 	/home/runner/work/focalboard/focalboard/server/app/blocks.go:60 +0x358  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971753+05:30] github.com/mattermost/focalboard/server/api.(*API).handlePostBlocks(0xc00054e600, 0x1f46880, 0xc000aa73c0, 0xc0009a0900)  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.9717681+05:30] 	/home/runner/work/focalboard/focalboard/server/api/api.go:343 +0x2ad  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.9717739+05:30] github.com/mattermost/focalboard/server/api.(*API).attachSession.func1(0x1f46880, 0xc000aa73c0, 0xc0009a0800)  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.9717782+05:30] 	/home/runner/work/focalboard/focalboard/server/api/auth.go:387 +0x9cd  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.9717839+05:30] net/http.HandlerFunc.ServeHTTP(0xc000537f00, 0x1f46880, 0xc000aa73c0, 0xc0009a0800)  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971787+05:30] 	/opt/hostedtoolcache/go/1.15.13/x64/src/net/http/server.go:2042 +0x44  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971792+05:30] github.com/mattermost/focalboard/server/api.(*API).requireCSRFToken.func1(0x1f46880, 0xc000aa73c0, 0xc0009a0800)  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971801+05:30] 	/home/runner/work/focalboard/focalboard/server/api/api.go:104 +0x1c4  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971807+05:30] net/http.HandlerFunc.ServeHTTP(0xc000129400, 0x1f46880, 0xc000aa73c0, 0xc0009a0800)  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.97181+05:30] 	/opt/hostedtoolcache/go/1.15.13/x64/src/net/http/server.go:2042 +0x44  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.9718149+05:30] github.com/gorilla/mux.(*Router).ServeHTTP(0xc0005640c0, 0x1f46880, 0xc000aa73c0, 0xc000204500)  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.9718258+05:30] 	/home/runner/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0xd3  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971838+05:30] main.(*Plugin).ServeHTTP(0xc00007e5f0, 0xc000b9e420, 0x1f46880, 0xc000aa73c0, 0xc000204500)  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971843+05:30] 	/home/runner/work/focalboard/focalboard/mattermost-plugin/server/plugin.go:130 +0x6c  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.9718568+05:30] github.com/mattermost/mattermost-server/v5/plugin.(*hooksRPCServer).ServeHTTP(0xc000468360, 0xc000108e20, 0x26582f8, 0x0, 0x0)  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971862+05:30] 	/home/runner/go/pkg/mod/github.com/mattermost/mattermost-server/v5@v5.3.2-0.20210524045451-a4f7df6f6e3c/plugin/client_rpc.go:366 +0x2b2  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.9718919+05:30] reflect.Value.call(0xc0000431a0, 0xc0000b6808, 0x13, 0x1dbb657, 0x4, 0xc0000a3f08, 0x3, 0x3, 0x1bc3c00, 0xc000574630, ...)  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971898+05:30] 	/opt/hostedtoolcache/go/1.15.13/x64/src/reflect/value.go:476 +0x8c7  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971921+05:30] reflect.Value.Call(0xc0000431a0, 0xc0000b6808, 0x13, 0xc0004f9f08, 0x3, 0x3, 0x1f4bb00, 0x2000000000000, 0x2000100000001)  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971937+05:30] 	/opt/hostedtoolcache/go/1.15.13/x64/src/reflect/value.go:337 +0xb9  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971966+05:30] net/rpc.(*service).call(0xc000202000, 0xc00007e0f0, 0xc00043c160, 0xc00043c170, 0xc000136c00, 0xc000394240, 0x1baf740, 0xc000108e20, 0x16, 0x1bc6000, ...)  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971972+05:30] 	/opt/hostedtoolcache/go/1.15.13/x64/src/net/rpc/server.go:377 +0x189  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.971981+05:30] created by net/rpc.(*Server).ServeCodec       caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
DEBU[2021-06-30T12:01:42.9719849+05:30] 	/opt/hostedtoolcache/go/1.15.13/x64/src/net/rpc/server.go:474 +0x445  caller="plugin/hclog_adapter.go:54" plugin_id=focalboard
ERRO[2021-06-30T12:01:42.973952+05:30] Plugin failed to ServeHTTP, RPC call failed   caller="mlog/log.go:251" error="unexpected EOF" plugin_id=focalboard
ERRO[2021-06-30T12:01:42.974014+05:30] Plugin failed to ServeHTTP, RPC call failed   caller="mlog/log.go:251" error="unexpected EOF" plugin_id=focalboard
ERRO[2021-06-30T12:01:42.9740002+05:30] Plugin failed to ServeHTTP, RPC call failed   caller="mlog/log.go:251" error="unexpected EOF" plugin_id=focalboard
ERRO[2021-06-30T12:01:42.974095+05:30] Plugin failed to ServeHTTP, RPC call failed   caller="mlog/log.go:251" error="unexpected EOF" plugin_id=focalboard
ERRO[2021-06-30T12:01:42.974241+05:30] http: response.WriteHeader on hijacked connection from github.com/mattermost/mattermost-server/v5/plugin.(*hooksRPCClient).ServeHTTP (client_rpc.go:418)  caller="http/server.go:3137" source=httpserver
ERRO[2021-06-30T12:01:42.974289+05:30] http: response.Write on hijacked connection from fmt.Fprintln (print.go:265)  caller="http/server.go:3137" source=httpserver
DEBU[2021-06-30T12:01:42.974357+05:30] plugin process exited                         caller="plugin/hclog_adapter.go:52" plugin_id=focalboard wrapped_extras="pathplugins/focalboard/server/dist/plugin-darwin-amd64pid17939errorexit status 2"
ERRO[2021-06-30T12:01:47.421668+05:30] Plugin failed to ServeHTTP, RPC call failed   caller="mlog/log.go:251" error="connection is shut down" plugin_id=focalboard
DEBU[2021-06-30T12:01:47.47822+05:30] Received HTTP request                         caller="web/handlers.go:105" method=GET request_id=oqr8a81euiygxy5i1gbkbajpfo status_code=200 url=/favicon.ico
WARN[2021-06-30T12:01:48.5981+05:30] Health check failed for plugin                caller="plugin/health_check.go:59" error="plugin RPC connection is not responding" id=focalboard
DEBU[2021-06-30T12:01:48.598161+05:30] Deactivating plugin due to multiple crashes   caller="plugin/health_check.go:65" id=focalboard
ERRO[2021-06-30T12:01:48.5981739+05:30] RPC call OnDeactivate to plugin failed.       caller="mlog/log.go:251" error="connection is shut down" plugin_id=focalboard
WARN[2021-06-30T12:01:48.598204+05:30] error closing client during Kill              caller="plugin/hclog_adapter.go:70" plugin_id=focalboard wrapped_extras="errconnection is shut down"
WARN[2021-06-30T12:01:48.598221+05:30] plugin failed to exit gracefully              caller="plugin/hclog_adapter.go:72" plugin_id=focalboard
```

**Comment by harshilsharma63** [comment]
The root cause - `panic: concurrent write to websocket connection`

**Comment by harshilsharma63** [comment]
This is fixed with #658
